pkg/deploy/util/util.go

```diff
@@ -390,6 +397,35 @@ func int32AnnotationFor(obj runtime.Object, key string) (int32, bool) {
 	return int32(i), true
 }
 
+// DeploymentsForCleanup determines which deployments for a configuration are relevant for the
+// revision history limit quota
+func DeploymentsForCleanup(configuration *deployapi.DeploymentConfig, deployments []api.ReplicationController) []api.ReplicationController {
+	// if the past deployment quota has been exceeded, we need to prune the oldest deployments
+	// until we are not exceeding the quota any longer, so we sort oldest first
+	sort.Sort(ByLatestVersionAsc(deployments))
+
+	activeDeployment := ActiveDeployment(configuration, deployments)
+	if activeDeployment == nil {
+		// if cleanup policy is set but no successful deployments have happened, there will be
+		// no active deployment. We can consider all of the deployments in this case except for
+		// the latest one
+		return deployments[:len(deployments)-1]
```

Comment:
The sort above accomplishes this, and it only exists there for the express purpose of this line